src/lib/hooks/useErrorHandler.ts

```diff
@@ -0,0 +1,34 @@
+import { Errors } from "../domain";
+import { useMountedOnlyState } from "./useMountedOnlyState";
+
+type HookResult = {
+    errorHandler: <T>(operation: Promise<T>) => Promise<T | undefined>,
+    errors: Array<Error>,
+    clearError: (error: Error) => void
+};
+
+export function useErrorHandler(): HookResult {
+    const [errors, setErrors] = useMountedOnlyState<Errors>({});
+
+    return {
+        errorHandler: async <T>(operation: Promise<T>) => {
+            try {
+                return await operation;
+            }
+            catch (error) {
+                setErrors(currentErrors => {
+                    const message = error instanceof Error ? error?.message : error;
+                    return {
+                        ...currentErrors,
+                        [message]: { message }
```

Comment:
Include the rest of the error when the given error is an object. Test that the rest of the object is available from the errors property of the hook return.